src/cgroups/cpu.rs

```diff
@@ -0,0 +1,88 @@
+//!
+//! Reference: https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#cpu
+
+use crate::{
+    cgroups::Controller,
+    rootfs::default_devices,
+    spec::{LinuxCpu, LinuxDeviceType, LinuxResources},
+};
+
+use anyhow::Result;
+use nix::unistd::Pid;
+use std::path::Path;
+
+pub struct Cpu {}
+
+impl Controller for Cpu {
+    fn apply(linux_resources: &LinuxResources, cgroup_root: &Path, pid: Pid) -> Result<()> {
+        fs::create_dir_all(cgroup_root)?;
+
+        if let Some(cpu) = &linux_resources.cpu {
+            Self::apply(linux_resources, cgroup_root, pid)
```

Comment:
Is this method declared? It looks like you have method `apply_cpu`. Otherwise it looks like you're trying to recursively declare the method that's being implemented, and I don't think this is allowed in a trait implementation. Inside a trait implementation calls to `Self::` methods won't consider the trait being implemented since it's not considered to be implemented yet as you're currently implementing it. Also this isn't really what you probably want to be doing, you should be invoking your `apply_cpu` method here.